google-cloud-spanner/src/test/java/com/google/cloud/spanner/OpenTelemetrySpanTest.java

```diff
@@ -0,0 +1,262 @@
+        StatementResult.exception(
+            INVALID_UPDATE_STATEMENT,
+            Status.INVALID_ARGUMENT.withDescription("invalid statement").asRuntimeException()));
+    String uniqueName = InProcessServerBuilder.generateName();
+    server =
+        InProcessServerBuilder.forName(uniqueName)
+            // We need to use a real executor for timeouts to occur.
+            .scheduledExecutorService(new ScheduledThreadPoolExecutor(1))
+            .addService(mockSpanner)
+            .build()
+            .start();
+
+    channelProvider = LocalChannelProvider.create(uniqueName);
+  }
+
+  @AfterClass
+  public static void stopServer() throws InterruptedException {
+    if (server != null) {
+      server.shutdown();
+      server.awaitTermination();
+    }
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    spanExporter = InMemorySpanExporter.create();
+
+    SdkTracerProvider tracerProvider =
+        SdkTracerProvider.builder()
+            .addSpanProcessor(SimpleSpanProcessor.create(spanExporter))
+            .build();
+
+    OpenTelemetry openTelemetry =
+        OpenTelemetrySdk.builder()
+            .setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()))
+            .setTracerProvider(tracerProvider)
+            .build();
+
+    SpannerOptions.Builder builder =
+        SpannerOptions.newBuilder()
+            .setProjectId(TEST_PROJECT)
+            .setChannelProvider(channelProvider)
+            .setCredentials(NoCredentials.getInstance())
+            .setOpenTelemetry(openTelemetry)
+            .setSessionPoolOption(SessionPoolOptions.newBuilder().setMinSessions(0).build());
+
+    spanner = builder.build().getService();
+
+    client = spanner.getDatabaseClient(DatabaseId.of(TEST_PROJECT, TEST_INSTANCE, TEST_DATABASE));
+  }
+
+  @After
+  public void tearDown() {
+    spanner.close();
+    mockSpanner.reset();
+    mockSpanner.removeAllExecutionTimes();
+    spanExporter.reset();
+  }
+
+  @Test
+  public void singleUse() {
+    try (ResultSet rs = client.singleUse().executeQuery(SELECT1)) {
+      while (rs.next()) {
+        // Just consume the result set.
+      }
+      List<String> spans =
+          spanExporter.getFinishedSpanItems().stream()
+              .map(SpanData::getName)
+              .distinct()
+              .collect(Collectors.toList());
+
+      boolean hasSpanData =
+          spans.stream()
+              .allMatch(
+                  name ->
+                      name.equals("CloudSpannerOperation.BatchCreateSessionsRequest")
+                          || name.equals("CloudSpannerOperation.ExecuteStreamingQuery")
+                          || name.equals("CloudSpannerOperation.BatchCreateSessions")
+                          || name.equals("CloudSpanner.ReadOnlyTransaction")
+                          || name.equals("SessionPool.WaitForSession"));
+      assertTrue(hasSpanData);
+      assertThat(spans.size()).isEqualTo(5);
```

Comment:
nit (here and elsewhere): prefer `assertEquals` over `assertThat(...).isEqualTo(...)`